File: leetcode/leetcode-226.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        def reverse(u):
            if u is None:
                return
            u.left, u.right = u.right, u.left
            reverse(u.left)
            reverse(u.right)
        reverse(root)
        return root

File: leetcode/leetcode-226.cpp
/*
 *@File Name          :leetcode-226.cpp
 *@Author             :cc
 *@Version            :1.0.0
 *@Date               :2022/7/18 10:03
 *@Description        :
 *@Function List      :
 *@History            :
 */

#include "algorithm"
using namespace std;

struct TreeNode {
  int       val;
  TreeNode* left;
  TreeNode* right;
  TreeNode() : val(0), left(nullptr), right(nullptr) {}
  TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
  TreeNode(int x, TreeNode* left, TreeNode* right)
      : val(x), left(left), right(right) {}
};

class Solution {
 public:
  TreeNode* invertTree(TreeNode* root) {
    if (root == nullptr) return nullptr;
    swap(root->left, root->right);
    invertTree(root->right);
    invertTree(root->left);
    return root;
  }
};